src/controller/datacontroller.js

```diff
@@ -306,6 +335,57 @@ export default class DataController {
 
 mix( DataController, EmitterMixin );
 
+// Traverses given DocumentFragment and searches elements which marks marker range. Founded element is removed from
+// DocumentFragment but path of this element is stored in Map.
+//
+// @param {module:engine/view/documentfragment~DocumentFragment} documentFragment Model DocumentFragment.
+// @returns {Object} Object with markers data and cleaned up document fragment.
+function extractMarkersFromModelFragment( documentFragment ) {
```

Comment:
> Also, since documentFragment is passed by reference, you only need to return the markers. It will be more elegant to use.

That's how it looked in my first implementation but I assumed that it'll be more readable to return data.